normandy/control/static/control/js/components/ActionForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { reduxForm } from 'redux-form'
+import { _ } from 'underscore'
+import HeartbeatForm from './action_forms/HeartbeatForm.jsx'
+import ConsoleLogForm from './action_forms/ConsoleLogForm.jsx'
+
+class ActionForm extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const currentProps = _.pick(this.props, 'initialValues', 'values', 'fields', 'selectedAction');
+    const incomingProps = _.pick(nextProps, 'initialValues', 'values', 'fields', 'selectedAction');
+
+    return (!_.isEqual(currentProps, incomingProps) || !_.isEqual(this.state, nextState));
```

Comment:
Good question. I can take out `initialValues` and `selectedAction`, and replace them with `arguments_schema`. I still need `values` because the input values are set to `values[inputName]`, and we need a re-render in order to see the changes we make to each field. (If you take `values` out, you can type in an input field and it will look as if nothing is happening.)

Without this, re-rendering occurs an INSANE amount of times in redux-forms. It fires after literally anything you could possibly do with your mouse or keyboard (onBlur, for example) because there are so many things changing in the form state at any given time.  It just seemed excessive and unnecessary. Also, any time the recipeForm re-rendered, the actionForm would go along for a ride as well. It was definitely laggy while I had devtools opened, but that could've been devtools-specific. I can test it out and see how it works if I remove these lifecycle methods.